docs/api/platform/flow_control.md

```diff
@@ -2,51 +2,68 @@
 
 We can use Blinky to explore flow control and task management in Arm Mbed OS applications. We'll look at automated actions first, then move on to handling user actions.
 
-## Flow control for automated actions
+## Flow control for delayed actions
 
-If we want to automatically blink an LED, we have three main techniques:
+If we want to automatically blink an LED, we have four main techniques:
 
-1. [Wait](#wait)
-1. [Ticker](#ticker)
+1. [Busy wait](#busy-wait)
+1. [Ticker, Timeout](#ticker-timeout)
 1. [Thread](#thread)
+1. [EventQueue](#eventqueue)
 
-### Wait
+All above techniques accomplish delays but cater to different requirements for precision, efficiency and context.
 
-#### Busy wait
+<span class="tips">**Tip:** You may want to read the [power optimization](../apis/platform-concepts.html) tutorial to understand how to achieve power savings. </span>
+
+### Busy wait
 
-Busy wait is a method that blocks the processor for a period of time. This is an effective way to create time delays, but it is inefficient because it wastes processor time and keeps the processor running at full power for the duration of the wait:
+Busy wait is a method that blocks the processor for a period of time. The processor runs at full power for the duration of the wait:
 
 [![View code](https://www.mbed.com/embed/?url=https://github.com/ARMmbed/mbed-os-snippet-Flow-Control-Busy-Wait)](https://github.com/ARMmbed/mbed-os-snippet-Flow-Control-Busy-Wait/blob/v6.0/main.cpp)
 
-Notice `printf()`; you can enable this by uncommenting the line (remove the `//`). `printf()` prints to the terminal, so you can use it to get debug information.
+Notice `wait_us()` - it is part of the [Wait API](../mbed-os-api-doxy/group__platform__wait__api.html) to busy wait a given number of microseconds.
 
-<span class="tips">**Tip:** Busy wait is inefficient, and we recommend not using it. </span>
+The Wait API is an ISR-safe way to create short delays of nanoseconds to a few microseconds. However, we do not recommend busy waiting for longer delays.
```

Comment:
What's ISR?

Reply:
Again, because I'm not sure what I'm trying to achieve with these delays, I don't know why I'll want to use busy wait.

Reply:
This is talking about the Wait API and busy wait as if they're one and the same. Is this true?

Reply:
Will improve this